src/pybind/mgr/cephadm/module.py

```diff
@@ -3985,6 +4000,47 @@ def remove_osds_status(self) -> List[OSD]:
         """
         return self.to_remove_osds.all_osds()
 
+    @handle_orch_error
+    @service_exists()
+    def set_osd_spec(self, service_name: str, osd_ids: List[str]) -> str:
+        """
+        Update unit.meta file for osd with service name
+        """
+        daemons: List[orchestrator.DaemonDescription] = self.cache.get_daemons_by_type('osd')
+        update_osd = defaultdict(list)
+        for daemon in daemons:
+            if daemon.daemon_id in osd_ids:
+                if not daemon.hostname or not daemon.daemon_name:
+                    continue
+                update_osd[daemon.hostname].append(daemon.daemon_name)
+
+        if not update_osd:
+            raise OrchestratorError(f"Unable to find OSDs: {osd_ids}")
```

Comment:
is this actually printing out the ones we couldn't find or just printing out all the osd ids the user passed in if at least one can't be found?

Reply:
If none of the user passed osd ids is found then its printing all